### PR TITLE
feat: #225 add transaction pruning service

### PR DESCRIPTION
**Pull request by Sotatek-HuyLe3a**
#255

### CODE REVIEW

File: stores/transaction/src/main/java/com/bloxbean/cardano/yaci/store/transaction/scheduler/TransactionPruningService.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    private final TransactionStoreProperties transactionStoreProperties;
+
+    private final AtomicBoolean isPruning = new AtomicBoolean(false);
+
+    @PostConstruct
+    public void init() {
+        log.info("<< Transaction Pruning Service Enabled >>");
+    }
+
+    @Scheduled(fixedRateString = "${store.transaction.pruning-interval:86400}", timeUnit = TimeUnit.MINUTES)
```
Comment (matiwinnetou) [inline stores/transaction/src/main/java/com/bloxbean/cardano/yaci/store/transaction/scheduler/TransactionPruningService.java:41]: My suggestion is to keep the default or around 40 hours.

Comment (Sotatek-HuyLe3a) [reply stores/transaction/src/main/java/com/bloxbean/cardano/yaci/store/transaction/scheduler/TransactionPruningService.java:41]: So you mean we should set default is 40 hours, right?

Comment (satran004) [reply stores/transaction/src/main/java/com/bloxbean/cardano/yaci/store/transaction/scheduler/TransactionPruningService.java:41]: @matiwinnetou The current default 2160 (43200 slots) blocks is taken from the protocol parameter "securityParam" as no rollback possible after this duration. We are using the same value for other stores like utxo.

But I think, the application will set their own safe  slots based on application requirement.

File: stores/transaction/src/main/java/com/bloxbean/cardano/yaci/store/transaction/TransactionStoreProperties.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class TransactionStoreProperties {
+
+    @Builder.Default
+    private boolean pruningEnabled = false;
+
+    @Builder.Default
+    private int pruningInterval = 144000;
```
Comment (satran004) [inline stores/transaction/src/main/java/com/bloxbean/cardano/yaci/store/transaction/TransactionStoreProperties.java:18]: @Sotatek-HuyLe3a Let's keep this interval to 24hrs. 86400 sec.

File: stores/transaction/src/main/java/com/bloxbean/cardano/yaci/store/transaction/TransactionStoreProperties.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    @Builder.Default
+    private int pruningSafeSlot = 43200; // 2160 blocks
```
Comment (satran004) [inline stores/transaction/src/main/java/com/bloxbean/cardano/yaci/store/transaction/TransactionStoreProperties.java:21]: I think @matiwinnetou  was referring to this value. 
But let's keep it as it is, as we are using similar value in other stores and it's configurable.

File: stores/transaction/src/main/java/com/bloxbean/cardano/yaci/store/transaction/scheduler/TransactionPruningService.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    private final TransactionStoreProperties transactionStoreProperties;
+
+    private final AtomicBoolean isPruning = new AtomicBoolean(false);
+
+    @PostConstruct
+    public void init() {
+        log.info("<< Transaction Pruning Service Enabled >>");
+    }
+
+    @Scheduled(fixedRateString = "${store.transaction.pruning-interval:144000}", timeUnit = TimeUnit.MINUTES)
```
Comment (satran004) [inline stores/transaction/src/main/java/com/bloxbean/cardano/yaci/store/transaction/scheduler/TransactionPruningService.java:41]: Please change the time unit to second and the default value accordingly.

**Review by satran004 — APPROVED** [review]

**Pull request merged**